config/confighttp/server.go

```diff
@@ -167,8 +167,18 @@ func WithDecoder(key string, dec func(body io.ReadCloser) (io.ReadCloser, error)
 	})
 }
 
+// WithServerExtensions is a [ToServerOption] which supplies the map of extensions to search for middlewares.
+func WithServerExtensions(extensions map[component.ID]component.Component) ToServerOption {
```

Comment:
Let's add an example here on how this would be typically called (same as in your release note or what you say in `ToServer`)